crates/pathfinder/src/state/sync.rs

```diff
@@ -101,282 +100,226 @@ where
     ));
 
     // Start L1 and L2 sync processes.
-    let mut l1_handle = tokio::spawn(l1_sync(
-        tx_l1,
-        transport.clone(),
-        chain,
-        core_address,
-        l1_head,
-    ));
+    let poll_interval = head_poll_interval(chain);
+    tokio::spawn(l1_sync(tx_l1, ethereum_client.clone(), poll_interval));
+    let no_delay = std::time::Duration::ZERO;
     let mut l2_handle = tokio::spawn(l2_sync(
         tx_l2,
         sequencer.clone(),
         l2_head,
         chain,
         pending_poll_interval,
+        no_delay,
         block_validation_mode,
     ));
 
-    let mut existed = (0, 0);
-
-    let mut last_block_start = std::time::Instant::now();
-    let mut block_time_avg = std::time::Duration::ZERO;
-    const BLOCK_TIME_WEIGHT: f32 = 0.05;
-    /// Delay before restarting L1 or L2 tasks if they fail. This delay helps prevent DoS if these
-    /// tasks are crashing.
-    #[cfg(not(test))]
-    const RESET_DELAY_ON_FAILURE: std::time::Duration = std::time::Duration::from_secs(60);
-
     loop {
         tokio::select! {
-            l1_event = rx_l1.recv() => match l1_event {
-                Some(l1::Event::Update(updates)) => {
-                    let first = updates.first().map(|u| u.block_number.get());
-                    let last = updates.last().map(|u| u.block_number.get());
-
-                    l1_update(&mut db_conn, &updates).await.with_context(|| {
-                        format!("Update L1 state with blocks {first:?}-{last:?}")
-                    })?;
-
-                    match updates.as_slice() {
-                        [single] => {
-                            tracing::info!("L1 sync updated to block {}", single.block_number);
-                        }
-                        [first, .., last] => {
-                            tracing::info!(
-                                "L1 sync updated with blocks {} - {}",
-                                first.block_number,
-                                last.block_number
-                            );
-                        }
-                        _ => {}
+            l1_event = rx_l1.recv() => if let Some(update) = l1_event {
+                handle_l1_event(update, &mut db_conn)?;
```

Comment:
shouldn't this function be run in tokio::task::spawn_blocking or by using block_in_place to avoid [starving other branches](https://docs.rs/tokio/latest/tokio/macro.select.html#runtime-characteristics)?